src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1189,6 +1189,11 @@ private async Task HandleBreakModeEvent(ResultEventArgs results, BreakRequest br
                     bplist.AddRange(bkpt);
                     _callback.OnBreakpoint(thread, bplist.AsReadOnly());
                 }
+                else if (ExceptionManager.TryGetExceptionBreakpoint(bkptno, out EXCEPTION_INFO exceptionInfo) == Constants.S_OK) // exception breakpoint hit
+                {
+                    ExceptionBreakpointStates exceptionBreakpointStates = ExceptionManager.ToExceptionBreakpointState(exceptionInfo.dwState);
+                    _callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, exceptionBreakpointStates);
```

Comment:
>OnException [](http://example.com/codeflow?start=30&length=11)

nit: We should probably get more agreement between `TryGetExceptionBreakpoint` and `OnException` on the types used, at the least they should both probably agree on enum_EXCEPTION_STATE vs ExceptionBreakpointStates.

Reply:
> at the least they should both probably agree on enum_EXCEPTION_STATE vs ExceptionBreakpointStates.

Since `categorySettings.CategoryState` is always `BreakThrown` as you mentioned when `TryGetExceptionBreakpoint()` is called in the "breakpoint-hit" condition, we can remove the `exceptionInfo.dwState` assignment completely and replace

```
ExceptionBreakpointStates exceptionBreakpointStates = ExceptionManager.ToExceptionBreakpointState(exceptionInfo.dwState);
_callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, exceptionBreakpointStates);
```

with `_callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, ExceptionBreakpointStates.BreakThrown);`